src/megaapi_impl.cpp

```diff
@@ -13367,27 +13370,55 @@ void MegaApiImpl::syncupdate_local_lockretry(bool waiting)
 }
 #endif
 
-void MegaApiImpl::backupput_result(const Error&, handle backupId)
+void MegaApiImpl::backupput_result(const Error& e, handle backupId)
 {
+    if (e || backupId == UNDEF)
+    {
+        LOG_err << "backupput_result failed: " << MegaError::getErrorString(e);
+    }
+
+    if (requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if (!request || (request->getType() != MegaRequest::TYPE_BACKUP_PUT)) return;
+    request->setParentHandle(backupId);
     mHeartBeatMonitor->digestPutResult(backupId);
+    fireOnRequestFinish(request, make_unique<MegaErrorPrivate>(e));
 }
 
-void MegaApiImpl::backupupdate_result(const Error&, handle /*backupId*/)
+void MegaApiImpl::backupupdate_result(const Error& e, handle backupId)
 {
+    if (e || backupId == UNDEF)
+    {
+        LOG_err << "backupupdate_result failed: " << MegaError::getErrorString(e);
+    }
 
+    if (requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if (!request || (request->getType() != MegaRequest::TYPE_BACKUP_PUT)) return;
+    request->setParentHandle(backupId);
```

Comment:
For updates, it's already set as input param. If you want to, we can add an assert, but not really needed to re-assign it.
```suggestion
    assert(backupId == request->getParentHandle()
```